File: Python/Unique_Length-3_Palindromic_Subsequences.py
# 1930. Unique Length-3 Palindromic Subsequences

from operator import le


def countPalindromicSubsequence(self, s: str) -> int:
    res = 0
    unq_str = set(s)
    for ch in unq_str:
        left = s.find(ch)
        right = s.rfind(ch)

        if(right > left):
            res += len(set(s[left + 1: right]))

    return res

File: JavaScript/unique-length-3-palindromic-subsequences.js
// 1930. Unique Length-3 Palindromic Subsequences

var countPalindromicSubsequence = function(s) {
    const unique_chars = [...new Set(s.split(''))];
    
    return unique_chars.reduce((count, ch)=>{
        const first_one = s.indexOf(ch);
        const last_one = s.lastIndexOf(ch);
        if (last_one > first_one)
            return count + new Set(s.slice(first_one + 1, last_one)).size
        else return count
    }, 0)   
};